src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -460,6 +460,57 @@ class HttpGoogleServicesDAO(
     }
   }
 
+  def getBilling(credential: Credential) = {
+    new Cloudbilling.Builder(httpTransport, jsonFactory, credential).build()
+  }
+
+  /**
+    * NOTE: This function will returns "false" in both of the following cases:
+    * if you don't have sufficient scopes
+    *   - Google's JSON response body will contain "message": "Request had insufficient authentication scopes."
+    * if you're not authorized to see the billing account
+    *   - Google's JSON response body will contain "message" : "The caller does not have permission"
+    */
+  protected def credentialOwnsBillingAccount(credential: Credential, billingAccountName: String): Future[Boolean] = {
+    val fetcher = getBilling(credential).billingAccounts().get(billingAccountName)
+    retryWithRecoverWhen500orGoogleError(() => {
+      blocking {
+        executeGoogleRequest(fetcher)
+      }
+      true //if the request succeeds, it has access.
+    }) {
+      case e: HttpResponseException if e.getStatusCode == StatusCodes.Forbidden.intValue => false
+    }
+  }
+
+  protected def listBillingAccounts(credential: Credential): Future[Seq[BillingAccount]] = {
+    val fetcher = getBilling(credential).billingAccounts().list()
+    retryWithRecoverWhen500orGoogleError(() => {
+      val list = blocking {
+        executeGoogleRequest(fetcher)
+      }
+      list.getBillingAccounts.toSeq
+    }) {
+      case e: HttpResponseException if e.getStatusCode == StatusCodes.Forbidden.intValue =>
+        throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, e.getContent))
```

Comment:
I agree this needs to return a response indicating which scope(s) are required. Especially as we add more APIs that require specific scopes, the calling code in UI+orch needs to be somewhat generic; we shouldn't ask orchestration to know which scopes are required for which API calls. That knowledge is well-encapsulated here in rawls, so rawls should message it back to the caller. Possibly this means adding a new field to ErrorReport, or overriding one of the existing fields.

It sucks that you'd have to parse Google's response - but that's all the more reason to encapsulate it here.